components/bms_worker/src/CAN/CANIO-tx.c

```diff
@@ -202,66 +203,127 @@ static const packTable_S* packNextMessage(const packTable_S* packTable,
     return NULL;
 }
 
-static bool MSG_pack_BMS_100Hz(CAN_data_T* message, const uint8_t counter)
+static bool MSG_pack_BMS_100Hz_Critical(CAN_data_T* message, const uint8_t counter)
+{
+    message->u64  = ((uint64_t)((uint16_t)(BMS.voltage.min / 50) & 0x3FF)) << 54;   //10 bits, 5mV precision, range [0-5115]mV
```

Comment:
calculation is wrong. say 4/50 = 0.04 which as an int is 0. you need to work the other way ie voltage *1000/5. *1000 takes volts to mv, and /5 gives it on 10 bit size. also the range is the valid range, and the valid range is 0-5000